Src/Test.UnitTests.Sarif.PatternMatcher/PatternInvariantTests.cs

```diff
@@ -109,50 +162,182 @@ public static void VerifyAllRuleFilenamesMatchDefinitions(string definitionsFile
+            }
+
+            var outputmsg = new StringBuilder();
+            if (conflictingRuleIDList.Count > 0)
+            {
+                outputmsg.Append("These rules have multiple conflicting rule IDs issued for them" +
+                    $"{Environment.NewLine}  " +
+                    string.Join($",{Environment.NewLine}  ", conflictingRuleIDList.Distinct()) +
+                    $"{Environment.NewLine}");
+            }
+
+            if (sharedRuleIDList.Count > 0)
+            {
+                outputmsg.Append("These rule IDs have multiple conflicting rules issued for them" +
+                    $"{Environment.NewLine}  " +
+                    string.Join($",{Environment.NewLine}  ", sharedRuleIDList.Distinct()));
+            }
+
 
+            Assert.True(conflictingRuleIDList.Count == 0 && sharedRuleIDList.Count == 0, outputmsg.ToString());
         }
 
+        /// <summary>
+        /// This function ensures that for each regex in the JSON, there is a corresponding definition in the shared strings.
+        /// </summary>
         public static void VerifyAllSharedStringsExist(string definitionsFilePath, string sharedStringsFilePath)
         {
             string content = File.ReadAllText(definitionsFilePath);
-
             SearchDefinitions sdObject = JsonConvert.DeserializeObject<SearchDefinitions>(content);
 
             HashSet<string> regexSet = GetRegexSetFromSearchDefinitions(sdObject);
 
-            var rulesWithoutSharedStrings = new List<string>();
+            var regexesWithoutSharedStrings = new List<string>();
             string sharedStringsContents = File.ReadAllText(sharedStringsFilePath);
-            foreach (string rule in regexSet)
+
+            foreach (string regex in regexSet)
             {
-                if (rule != null && !sharedStringsContents.Contains(rule))
+                if (regex != null && !sharedStringsContents.Contains(regex))
                 {
-                    rulesWithoutSharedStrings.Add(rule);
+                    regexesWithoutSharedStrings.Add(regex);
                 }
             }
 
             // Assert.Empty doesn't allow custom messages, so use Assert.True
-            Assert.True(rulesWithoutSharedStrings.Count == 0,
+            Assert.True(regexesWithoutSharedStrings.Count == 0,
                         "Unable to find shared strings for these regular expression variables: " +
                         $"{Environment.NewLine}  " +
-                        string.Join($",{Environment.NewLine}  ", rulesWithoutSharedStrings));
+                        string.Join($",{Environment.NewLine}  ", regexesWithoutSharedStrings));
         }
 
+        /// <summary>
+        /// This function ensures that for each validator in the validator assembly, 
+        /// there is a corresponding test in the test assembly, and vice versa.
+        /// </summary>
+        public static void VerifyAllTestsExist(Assembly validatorsAssembly, Assembly testsAssembly)
```

Comment:
`VerifyAllTestFilesExist`? The method name implies we're checking the methods in the files, not the file names.